addons/etcd/dataprotection/restore.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+set -ex
+
+export PATH="$PATH:$DP_DATASAFED_BIN_PATH"
+export DATASAFED_BACKEND_BASE_PATH="$DP_BACKUP_BASE_PATH"
+
+DATA_DIR=/data
+mkdir -p ${DATA_DIR}
+res=$(find ${DATA_DIR} -type f)
+data_protection_file=${DATA_DIR}/.kb-data-protection
+if [ ! -z "${res}" ] && [ ! -f ${data_protection_file} ]; then
+  echo "${DATA_DIR} is not empty! Please make sure that the directory is empty before restoring the backup."
+  exit 1
+fi
+
+touch ${data_protection_file}
+
+remoteBackupFile="${DP_BACKUP_NAME}.tar.zst"
+if [ "$(datasafed list ${remoteBackupFile})" = "${remoteBackupFile}" ]; then
+  datasafed pull -d zstd-fastest "${remoteBackupFile}" - | tar -xvf - -C ${DATA_DIR}
+else
+  datasafed pull "${DP_BACKUP_NAME}.tar.gz" - | tar -xzvf - -C ${DATA_DIR}
```

Comment:
why the remote backup file name will be "${DP_BACKUP_NAME}.tar.gz"

Reply:
cleanup